### PR TITLE
Rework RuntimeFieldType class and direct subclass

### PR DESCRIPTION
**Pull request by javanna**
RuntimeFieldType is currently light and most of its concrete implementation is in AbstractScriptFieldType. Now that they both live in the same module, some members that don't have to do with scripting can be pushed to the base class RuntimeFieldType for other implementors to use.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java [hunk 1]
```diff
@@ -39,15 +63,194 @@ public final XContentBuilder toXContent(XContentBuilder builder, Params params)
...
+    }
+
+    @Override
+    public final boolean isSearchable() {
+        return true;
+    }
+
+    @Override
+    public final boolean isAggregatable() {
+        return true;
```
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java:77]: This is the case for all current implementations but not necessarily for all future ones? I think `isSearchable` and `isAggregatable` should probably live in the same class as the concrete methods that implement searching and fielddata

**Review by romseygeek — APPROVED** [review]
One nit, LGTM otherwise!

File: server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java (hunk 1 above)
Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java:77]: I do think that the point of runtime fields is that they are always searchable and aggregatable as they are calculated/loaded at runtime. I cannot think of a runtime fields that would not be searchable and aggregatable... can you?

Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/RuntimeFieldType.java:77]: I am leaving this as-is for now, always happy to revisit if needed.

**Pull request merged**